### ISSUE TITLE
[HOLD for payment 2023-05-03] [$2000] Sliding using fingers on android from pdf to image gets stuck and doesn't work the way how sliding from image to pdf works

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15988]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open android > Go to any chat > Send Image 1, Image 2, and finally a pdf file
2. Preview image 1, slide right using fingers (not the right arrow slider), and see that image 2 displays. Next, right-slide it again and see the pdf also displays.
3. Next, slide left from pdf and see that it gets stuck while swiping and sometimes it doesn't swipe too from pdf to the image. It works totally fine when we try to slide using fingers from any images to pdf.

## Expected Result:
Sliding using fingers from pdf to image gets stuck and sometimes doesn't swipe too and doesn't work  the way  how sliding from image to pdf works

## Actual Result:
Sliding using fingers from pdf to image gets stuck and doesn't work

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.85-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/225320544-8b3b9adc-195c-4ec4-a4af-c5cae37e52b1.mp4

https://user-images.githubusercontent.com/43996225/225320555-b380acc5-c487-427a-9273-07f617bbe0e0.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @avi-shek-jha
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1678863680773659?thread_ts=1678796475.385589&cid=C049HHMV9SM

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0181fbb67dfdeb58e8</li>
        <li>Upwork Job ID: 1638186266529497088</li>
        <li>Last Price Increase: 2023-04-11</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @alexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by alexpensify** [comment]
Catching up from being OOO and urgent bank GHs, I'll try to get to this one by EOW.

**Comment by alexpensify** [comment]
Ops GHs have been a higher priority, I haven't had time to test yet.

**Comment by alexpensify** [comment]
Focused on another urgent task, I'll try to get to this one tomorrow.

**Comment by luacmartins** [comment]
Just adding to the OP, swiping doesn't work at all when viewing PDF attachments -- it doesn't matter what the next attachment type is.

**Comment by alexpensify** [comment]
@luacmartins - looks like we see this as a problem. Do you think I should mark it as external? Thanks!

**Comment by luacmartins** [comment]
@alexpensify yes, it is a problem indeed and we can mark it external.

**Comment by luacmartins** [comment]
I'll assign myself as well to manage the engineering side since I was involved in the PR that implemented the carousel feature.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~0181fbb67dfdeb58e8

**Comment by MelvinBot** [comment]
Current assignee @alexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @0xmiroslav (`External`)

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by luacmartins** [comment]
@Ollyws's proposal looks good to me. What do you think @0xmiroslav ?

**Comment by luacmartins** [comment]
Still looking at proposals. Not overdue.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @JmillsExpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by alexpensify** [comment]
@JmillsExpensify - this one is moving forward with proposal reviews, but I need to reassign the `bug` label. I will be OOO until April 3 and need help keeping an eye out for any Bug team tasks when I'm offline. I have a note to step in if it's still open when I get back online.

**Comment by alexxxwork** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The carousel components doesn't swipe with single tap when displaying a PDF.

### What is the root cause of that problem?
The root cause of the problem is that PanResponder in AttachmentCarousel component intationally doesn't capture all gestures (onStartShouldSetPanResponderCapture, onMoveShouldSetPanResponderCapture aren't set)

### What changes do you think we should make in order to solve the problem?

First of all we should decide how this component should work. 
Right now it works so that PDF attachments doesn't swipe with one finger swipe but swipes with two finger swipe. So right now you could swipe from PDF to next attachment with two fingers but images attachments respond only on single tap swipe and there should be a consistent behaviour.

We could reverse this behaviour by setting onMoveShouldSetPanResponderCapture only when there's only one tap on screen like so:

```
onMoveShouldSetPanResponderCapture: (evt, gestureState) => gestureState.numberActiveTouches === 1
```
here 
https://github.com/Expensify/App/blob/8d34e12b337db7e32c2ae4d63a5af64796d5b1a2/src/components/AttachmentCarousel/CarouselActions/index.native.js#L28

That setting allows PDFView to respond on double tap to allow zooming and scrolling. And single tap swipe would be handled by GestureHandler allowing to navigate attachments.

This doesn't allow user to scroll PDF with a single tap when it's zoomed. To achive this we should know if we're in a zoomed state in upper component - CarouselActions where GestureHandler is set.

We could use onScaleChanged to get current scale and use it to prevent onMoveShouldSetPanResponderCapture when scale > 1
here:
https://github.com/Expensify/App/blob/8d34e12b337db7e32c2ae4d63a5af64796d5b1a2/src/components/PDFView/index.native.js#L156

setting onScaleChanged callback to get scale>1 cases and update our condition in CarouselActions:
```
onMoveShouldSetPanResponderCapture: (evt, gestureState) => !this.props.isZoomed && gestureState.numberActiveTouches === 1
```

**Comment by JmillsExpensify** [comment]
@alexpensify Sounds good! I'll take it from here.

**Comment by luacmartins** [comment]
This proposal seems good! @0xmiroslav any thoughts?

**Comment by alexxxwork** [comment]
> Your root cause is not correct because without setting `onStartShouldSetPanResponderCapture`, `onMoveShouldSetPanResponderCapture`, sliding already works on image view.

@0xmiroslav I've updated my proposal and explained the root cause more thoroghly, explaining why PDFView behaves differrently. Would you please take a look?

**Comment by MelvinBot** [comment]
@JmillsExpensify @luacmartins @0xmiroslav this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by luacmartins** [comment]
@0xmiroslav did you get a chance to look at the [revised proposal](https://github.com/Expensify/App/issues/15988#issuecomment-1482739389)?

**Comment by ahmedGaber93** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Sliding using fingers on android from pdf to image gets stuck.
### What is the root cause of that problem?
when you open pdf, the `PDF` component is the responser for the touch move event not `Carousel` component to can mange it's pages scroll.

### What changes do you think we should make in order to solve the problem?
on `Carousel` we will listen to move event. 
when user move `horzintally` the `Carousel` will be the responer to mange it's slides.
when user move `vertical` the `PDF` will be the responser to mange it's pdf pages scroll.

![Untitled](https://user-images.githubusercontent.com/41129870/229314910-c47a4137-179f-4bee-a5fa-fd6688f1ba9c.png)


but we may still have a problem when pdf zoomed and user swipe horzintally to see the pdf scaled area.
if it required, we need to tell `Carousel` to not be the responser for move event?.
so for it we can create new `PanResponder` in `PDFView` for the container View and use `event.preventDefault()`  and `event.defaultPrevented`
if pdf is zoomed, we will call `event.preventDefault()` 
and in `Carousel` will will check `event.defaultPrevented` to know if pdf zoomed or no see the code below.


```javascript
// in PDFView in the container View component
this.panResponder = PanResponder.create({
    onMoveShouldSetPanResponder: (event) => {
        if (this.state.isPdfScaled) {
            event.preventDefault();
        }
        return false; // always return false;
    },
});

// in `Carousel` we will check `event.defaultPrevented`
onMoveShouldSetPanResponder: (event, gestureState) => {
    if (Math.abs(gestureState.dx) > Math.abs(gestureState.dy) && !event.defaultPrevented) {
        return true;
    }
    return false;
},
```         

this the result




https://user-images.githubusercontent.com/41129870/229319686-6dc26cda-2698-46ca-994c-8a935cc6eb44.mov



### What alternative solutions did you explore? (Optional)

**Comment by ahmedGaber93** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/15988#issuecomment-1493165469)

**Comment by ahmedGaber93** [comment]
@0xmiroslav 
i think we can't allow to pdf to scroll vertically and curausol to scroll horzintally **at the same time** without handling `onMoveShouldSetPanResponder` between them.

```javascript
onMoveShouldSetPanResponder: (event, gestureState) => {
    if (Math.abs(gestureState.dx) > Math.abs(gestureState.dy)) {
        return true;
    }
    return false;
}, 
```

but i agree with you in pdf zoomed case the proposal is may be **workaround**.
A good solution is supposed to come from pdf library to `stopPropagation()` when pdf zoomed but it dosn't
so i suguest to warp the pdf component with a View and make it to `stopPropagation()` when pdf zoomed
```javascript
this.panResponder = PanResponder.create({
    onMoveShouldSetPanResponder: (event) => {
        if (this.state.scaled) {
            event.stopPropagation();
        }
        return false;
    },
});
```

**Comment by MelvinBot** [comment]
@JmillsExpensify @luacmartins @0xmiroslav this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a> and will need to go internal. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by JmillsExpensify** [comment]
Still working through proposals. We'll aim to get one approved and moving forward this week.

**Comment by MelvinBot** [comment]
@JmillsExpensify, @luacmartins, @0xmiroslav Huh... This is 4 days overdue. Who can take care of this?

**Comment by luacmartins** [comment]
[This solution](https://github.com/Expensify/App/issues/15988#issuecomment-1482739389) seems good to me. 

@0xmiroslav would you mind elaborating on why you think this is a workaround? I can see an argument for stopping propagation in the `react-native-pdf library` as mentioned [here](https://github.com/Expensify/App/issues/15988#issuecomment-1495169503), but I'm not sure that's the desired behavior in all instances which means that solution wouldn't get accepted by the library mantainers.

**Comment by MelvinBot** [comment]
@JmillsExpensify @luacmartins @0xmiroslav this issue is now 4 weeks old and preventing us from maintaining WAQ, can you:
- Decide whether any proposals currently meet our guidelines and can be approved as-is today
- If no proposals meet that standard, please take this issue internal and treat it as one of your highest priorities
- If you have any questions, don't hesitate to start a discussion in #expensify-open-source

Thanks!

**Comment by MelvinBot** [comment]
Current assignee @0xmiroslav is eligible for the Internal assigner, not assigning anyone new.

**Comment by alexxxwork** [comment]
> @alexxxwork's [proposal](https://github.com/Expensify/App/issues/15988#issuecomment-1482739389) does the job but I am thinking of [TouchableWithoutFeedback](https://github.com/Expensify/App/blob/ecc94d3f2e3af5739d7f6fa84a25b0805c2297ea/src/components/PDFView/index.native.js#L155) deprecation. It was introduced in #6169 and now redundant because #5870 is no longer reproducible. @alexxxwork can you please provide an updated proposal considering this?

@0xmiroslav Thanks for your input! [Updated ](https://github.com/Expensify/App/issues/15988#issuecomment-1482739389) proposal.

**Comment by ahmedGaber93** [comment]
@0xmiroslav 
What do you think of my [proposal](https://github.com/Expensify/App/issues/15988#issuecomment-1493165469) 👀, it works fine as you see in attachment video and my solution is very simple.

**Comment by alexxxwork** [comment]
> @alexxxwork does your updated proposal fix [#15988 (comment)](https://github.com/Expensify/App/issues/15988#issuecomment-1507272702)?

@0xmiroslav After some testing I should say that my proposal doesn't work well without TouchableWithoutFeedback.
The [issue ](https://github.com/Expensify/App/issues/5870) you mentioned is still reproducible if we change back Touchable to View.
This happens because we have several uses of TouchabeWithoutFeedback across the App and I think we should change them all to move to new API.
In this case we have two simultaneous TouchabeWithoutFeedback - one in PDFView and one in BasePreRenderer (which calls the context menu)

**Comment by ahmedGaber93** [comment]
@0xmiroslav 
>  is it safe to remove TouchabeWithoutFeedback with your solution?

before answer your question i need to understand why you need to remove it?

if we remove `TouchabeWithoutFeedback` the pdf scroll not work, because it prevent `Carousel` from to be the Responder, it's job is to **stop this line code below from work**.

https://github.com/Expensify/App/blob/8d34e12b337db7e32c2ae4d63a5af64796d5b1a2/src/components/AttachmentCarousel/CarouselActions/index.native.js#L28-L30


if we need to replace it with `View` we need to make the view to stopPropagation when touch.
```javascript
this.panResponder = PanResponder.create({
    onStartShouldSetPanResponder: (event) => {
        event.stopPropagation(); // here <<<<<<<<<<<<<<<<<<<<<<<
        return false;
    },
    onMoveShouldSetPanResponder: /*as i explained in the propsal*/,
});


<View {...this.panResponder.panHandlers}>
    <PDF />
</View>
```

but this soluation depond on my question above why we need to remove `TouchabeWithoutFeedback`? because my soluation may do the same behaviour of it.

**Comment by alexxxwork** [comment]
> if we need to replace it with `View` we need to make the view to stopPropagation when touch.

@ahmedGaber93 now zoom and try to tap and hold with two fingers. There shouldn't be a context menu in CarouselView.

**Comment by ahmedGaber93** [comment]
@alexxxwork yes There is no context menu displayed in CarouselView.

**Comment by ahmedGaber93** [comment]
@0xmiroslav 
> is it safe to remove TouchabeWithoutFeedback with your solution?

after i replay to you and sugguest solution [here](https://github.com/Expensify/App/issues/15988#issuecomment-1507737528)
please forget to tell me what you think of my proposal.

**Comment by MelvinBot** [comment]
@JmillsExpensify, @luacmartins, @0xmiroslav Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by luacmartins** [comment]
I think that @alexxxwork revised proposal addresses the concerns around `TouchabeWithoutFeedback`. Let's go with that.

**Comment by MelvinBot** [comment]
Current assignee @JmillsExpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Current assignee @0xmiroslav is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
📣 @alexxxwork You have been assigned to this job by @luacmartins!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by JmillsExpensify** [comment]
@alexxxwork Upwork job link is here: https://www.upwork.com/jobs/~0181fbb67dfdeb58e8. Please apply when you get a chance, and if the 50% bonus applies, then we will apply that on payout.

**Comment by alexxxwork** [comment]
@0xmiroslav @luacmartins The [PR](https://github.com/Expensify/App/pull/17647) is ready but there are some things to mention:
1. The logic of the AttachmentCarousel component changed significantly in recent PRs
2. So the logic of the proposal also changed a little.
3. Found some edge case introduced by this [PR](https://github.com/Expensify/App/pull/16973) - AttachmentCarousel now uses DeviceCapabilities.canUseTouchScreen() to identify whether we are on mobile or not. This leads to strange behaviour on laptops with touchscreen. Should we fix this in this PR? I can use isSmallScreenWidth or DeviceCapabilities.hasHoverSupport() for this.
Cc: @johnmlee101 @kidroca

**Comment by kidroca** [comment]
Hey guys, thanks for tagging me, but do we even have this bug after merging my PR?
- https://github.com/Expensify/App/pull/16973

I've updated the Carousel component to use a horizontal flat list that shows one image at a time 
A pan responder or a swipeable view is no longer necessary as all the swipe handling is covered by the FlatList 

I didn't notice any problems transitioning between PDFs and Images in the Carousel component when I tested my changes 

@alexxxwork what's the strange behavior that we need to fix?

`canUseTouchScreen` is partly responsible for enabling/disabling swipe navigation as it makes sense to only have that on touch devices. It's also used on a few places
`DeviceCapabilities.hasHoverSupport()` might work better for some of the usages, but I'm not sure whether it can cover all

**Comment by alexxxwork** [comment]
@kidroca First of all - great work on refactoring AttachmentCarousel and CarouselActions components - they are much cleaner now! 
On this issue we were investigating the problem with attachment swiping and came to the [solution](https://github.com/Expensify/App/issues/15988#issuecomment-1482739389) where user is able to change attachments with swipe and to zoom (pinch-zoom) and pan pdf attachments.
Although the swipe problem is now resolved there are still some issues on user experience - the pinch-zoom feature works with stuttering and you can't pan pdf attachment. 

The issue with a touch-screen laptop is that you can't scroll an attachment with mouse (as said in code comments `FlatList` prop `scrollEnabled` prevents this) and swipe gesture works in reversed manner (as said in code before `inverted` prop) so we could easily fix this for the cases when device has `hasHoverSupport()`.

**Comment by jasperhuangg** [comment]
Hey @alexxxwork looks like there was a regression associated with your PR. Looks like the zoom fails on mWeb for any PDFs with multiple pages. Please open a follow-up PR to fix this:

# mWeb - Chat - Zoom fails after pinch-zoom or double-tap on PDF and images

## Action Performed:
1. Go https://staging.new.expensify.com/
and login
2. Go to chat with attachments (pdf and images)
3. Tap any PDF/image
4. Pinch-zoom or double-tap to zoom

## Expected Result:
Attachment should be zoomed.

## Actual Result:
Zoom fails

**Comment by jasperhuangg** [comment]
@0xmiroslav got it, thanks for clarifying!

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.5-6 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17647


If no regressions arise, payment will be issued on 2023-05-03. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@0xmiroslav] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@0xmiroslav] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@0xmiroslav] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@0xmiroslav] Determine if we should create a regression test for this bug.
- [ ] [@0xmiroslav] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@JmillsExpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by JmillsExpensify** [comment]
Alright, let's get this issue kicked off. Looking at the linked PR, it appears that this PR does qualify for the 50% bonus based on this PR being merged three business days later. @0xmiroslav @luacmartins can you confirm? 

If that's the case, then we have the following payouts:
 - Issue reporter: $250 @avi-shek-jha
 - Contributor: $4,000 @alexxxwork 
 - Contributor+: $4,000 @0xmiroslav

**Comment by JmillsExpensify** [comment]
Then as for Upwork, the previous link has expired. @avi-shek-jha and @alexxxwork can both of you please apply ahead of the regression period ending?

**Comment by JmillsExpensify** [comment]
Finally, @0xmiroslav let's get the BugZero checklist kick off above. Thanks all!

**Comment by avi-shek-jha** [comment]
Seems like I can't access the slack link that you mentioned above. Here's my upwork profile link 
https://www.upwork.com/freelancers/~01a90e548bf808418c

It would be great if you could send me an invitation since I wasn't able to access both the upwork link and the slack link.

**Comment by JmillsExpensify** [comment]
Done! Sorry for the trouble.

**Comment by luacmartins** [comment]
@JmillsExpensify I assigned @alexxxwork to the PR on Apr 18 and we merged it on Apr 24, so I don't think this qualifies for the bonus based on our guidelines.

**Comment by JmillsExpensify** [comment]
@avi-shek-jha Great! We'll circle back on payments when the regression period passes. 

@luacmartins Great catch! I updated my comment above with the payouts.

**Comment by alexxxwork** [comment]
@JmillsExpensify Applied, thank you!

**Comment by JmillsExpensify** [comment]
Tomorrow we'll close this one out.

**Comment by JmillsExpensify** [comment]
Ah though @0xmiroslav before we do, please don't forget about the BugZero checklist, thank you!

**Comment by luacmartins** [comment]
@JmillsExpensify @0xmiroslav [raised a good point](https://expensify.slack.com/archives/D0533JSH674/p1682690090584529?thread_ts=1681973078.446999&cid=D0533JSH674) that issue took 6 days from assignment to PR being merged because another [PR](https://github.com/Expensify/App/pull/16973) completely changed the structure of the component the original proposal was based on. In light of that, I think that this issue qualifies for the bonus! Sorry about the back and forth!

**Comment by JmillsExpensify** [comment]
All good! I updated the original comment above with the correct amounts. I'm also issuing payments now.

**Comment by JmillsExpensify** [comment]
Payments issued to reporter and C+. Sent offer to contributor via Upwork! I'll pay the bonus when the contract is accepted and a final payment is made.

**Comment by JmillsExpensify** [comment]
@alexxxwork When you have a moment, please accept the Upwork offer/contract and we can make sure you are paid. Thank you!

**Comment by alexxxwork** [comment]
@JmillsExpensify Accepted, thank you!

**Comment by JmillsExpensify** [comment]
Thank you! Everyone is paid out and the regression test is linked above.
